Expired krb sessions don't fail the build

```
2019-05-01 01:27:34,395 cekit        DEBUG    Executing '/usr/bin/brew call --json-output listArchives checksum=cd68ad886a759d21f8dd0cb7646601ee type=maven'.
2019-05-01 01:27:35,812 [ERROR] koji: AuthError: unable to obtain a session
2019-05-01 01:27:35,858 cekit        ERROR    Can't fetch artifacts details from brew: 'b'''.
```

Resolved by running kinit again. I think it should fail hard here.

--- a/cekit/tools.py
+++ b/cekit/tools.py
@@ -64,25 +64,42 @@ def decision(question):
 
 def get_brew_url(md5):
     try:
-        LOGGER.debug("Getting brew details for an artifact with '%s' md5 sum" % md5)
+        LOGGER.debug("Getting brew details for an artifact with '{}' md5 sum".format(md5))
         list_archives_cmd = ['/usr/bin/brew', 'call', '--json-output', 'listArchives',
-                             'checksum=%s' % md5, 'type=maven']
-        LOGGER.debug("Executing '%s'." % " ".join(list_archives_cmd))
-        archive_yaml = yaml.safe_load(subprocess.check_output(list_archives_cmd))
+                             "checksum={}".format(md5), 'type=maven']
+        LOGGER.debug("Executing '{}'.".format(" ".join(list_archives_cmd)))
 
-        if not archive_yaml:
-            raise CekitError("Artifact with md5 checksum %s could not be found in Brew" % md5)
+        try:
+            json_archives = subprocess.check_output(list_archives_cmd).strip().decode("utf8")
+        except subprocess.CalledProcessError as ex:
+            if ex.output is not None and 'AuthError' in ex.output:
+                LOGGER.warning(
+                    "Brew authentication failed, please make sure you have a valid Kerberos ticket")
+            raise CekitError("Could not fetch archives for checksum {}".format(md5), ex)
 
-        archive = archive_yaml[0]
+        archives = yaml.safe_load(json_archives)
+
+        if not archives:
+            raise CekitError("Artifact with md5 checksum {} could not be found in Brew".format(md5))
+
+        archive = archives[0]
         build_id = archive['build_id']
         filename = archive['filename']
         group_id = archive['group_id']
         artifact_id = archive['artifact_id']
         version = archive['version']
 
-        get_build_cmd = ['brew', 'call', '--json-output', 'getBuild', 'buildInfo=%s' % build_id]
-        LOGGER.debug("Executing '%s'" % " ".join(get_build_cmd))
-        build = yaml.safe_load(subprocess.check_output(get_build_cmd))
+        get_build_cmd = ['brew', 'call', '--json-output',
+                         'getBuild', "buildInfo={}".format(build_id)]
+
+        LOGGER.debug("Executing '{}'".format(" ".join(get_build_cmd)))
+
+        try:
+            json_build = subprocess.check_output(get_build_cmd).strip().decode("utf8")
+        except subprocess.CalledProcessError as ex:
+            raise CekitError("Could not fetch build {} from Brew".format(build_id), ex)
+
+        build = yaml.safe_load(json_build)
 
         build_states = ['BUILDING', 'COMPLETE', 'DELETED', 'FAILED', 'CANCELED']
 

--- a/tests/test_unit_tools.py
+++ b/tests/test_unit_tools.py
@@ -1,5 +1,6 @@
 import logging
 import pytest
+import subprocess
 import yaml
 
 from contextlib import contextmanager
@@ -187,7 +188,7 @@ def brew_call_ok(*args, **kwargs):
             "artifact_id": "artifact_id",
             "version": "version",
           }
-        ]"""
+        ]""".encode("utf8")
     if 'getBuild' in args[0]:
         return """
         {
@@ -195,8 +196,8 @@ def brew_call_ok(*args, **kwargs):
           "release": "release",
           "state": 1
         }
-        """
-    return ""
+        """.encode("utf8")
+    return "".encode("utf8")
 
 
 def brew_call_removed(*args, **kwargs):
@@ -210,7 +211,7 @@ def brew_call_removed(*args, **kwargs):
             "artifact_id": "artifact_id",
             "version": "version",
           }
-        ]"""
+        ]""".encode("utf8")
     if 'getBuild' in args[0]:
         return """
         {
@@ -218,8 +219,8 @@ def brew_call_removed(*args, **kwargs):
           "release": "release",
           "state": 2
         }
-        """
-    return ""
+        """.encode("utf8")
+    return "".encode("utf8")
 
 
 def test_get_brew_url(mocker):
@@ -237,6 +238,22 @@ def test_get_brew_url_when_build_was_removed(mocker):
 
     assert 'Artifact with checksum aa was found in Koji metadata but the build is in incorrect state (DELETED) making the artifact not available for downloading anymore' in str(
         excinfo.value)
+
+
+# https://github.com/cekit/cekit/issues/502
+def test_get_brew_url_no_kerberos(mocker, caplog):
+    caplog.set_level(logging.DEBUG, logger="cekit")
+
+    kerberos_error = subprocess.CalledProcessError(1, 'CMD')
+    kerberos_error.output = "2019-05-06 14:58:44,502 [ERROR] koji: AuthError: unable to obtain a session"
+
+    mocker.patch('subprocess.check_output', side_effect=kerberos_error)
+
+    with pytest.raises(CekitError) as excinfo:
+        tools.get_brew_url('aa')
+
+    assert 'Could not fetch archives for checksum aa' in str(excinfo.value)
+    assert "Brew authentication failed, please make sure you have a valid Kerberos ticket" in caplog.text
 
 
 @contextmanager

